Libraries/VgVideo/vgKwaVideoClip.cxx

```diff
@@ -588,19 +590,23 @@ bool vgKwaVideoClip::resolvePadding(
     return false;
     }
 
-  CHECK_ARG(startTime <= endTime, false);
+  if (endTime < startTime && !qFuzzyCompare(startTime, endTime))
+    {
+    qWarning() << __func__ << "called with malformed times (end < start)";
+    return false;
+    }
```

Comment:
I'm struggling some with case where `endTime` and `startTime` are "equal", but actually `endTime` is slightly less than `startTime`.  Struggling in the sense that I'm wondering if something might go sideways elsewhere - but not obvious to me tht it would.  I wonder about making them equal if that is the case - but to what value?

Reply:
I expect that *probably* this wouldn't happen, but: [Robustness Principle](https://en.wikipedia.org/wiki/Robustness_principle).

What I think is actually happening that necessitated this PR is that time values are running through conversions somewhere that cause what we get from the the KWA and what we get from a query / saved results to be slightly different. This seems to be particularly an issue for single-frame results because any "fuzz" will cause us to have no overlap between the result and the KWA (whereas a multi-frame result might lose a frame but still be non-empty). I *expect* what we would normally see is that the start and end are bitwise identical (if off by maybe a few [ULPs](https://en.wikipedia.org/wiki/Unit_in_the_last_place) from the "correct" values), but the general approach has been to treat values within a few ULPs as equal.

I don't think there is much value making start and end equal here. If they're close but not bitwise equal, one or both of them is almost surely wrong, and so a) picking one would be arbitrary, and b) anywhere else will need to do fuzzy comparisons anyway.

Reply:
To be fair, I'm not 100% certain there aren't still places that should do fuzzing but don't; however, this seemed to fix the issues that we're seeing in VIAME.